orderable/models.py

```diff
@@ -67,6 +67,20 @@ def prev(self):
             sort_order__lt=self.sort_order
         ).order_by('-sort_order').first()
 
+    @staticmethod
+    def _update(qs):
+        """
+        Increment the sort_order in a queryset.
+
+        Handle IntegrityErrors caused by unique constraints.
+        """
+        try:
+            with transaction.atomic():
+                qs.update(sort_order=models.F('sort_order') + 1)
+        except IntegrityError:
+            for obj in qs.order_by('-sort_order'):
+                qs.filter(pk=obj.pk).update(sort_order=models.F('sort_order') + 1)
```

Comment:
Say we have a `qs` where `sort_order` looks like this: `1, 2, 3, 4`.
We add a new item with `sort_order == 2`.
`Orderable._save` creates `to_shift` which looks like: `2, 3, 4`.
The old code then tried to call `to_shift.update(sort_order = F('sort_order') + 1)`.
This works fine without any `unique` constraints, but if `sort_order` is `unique_together` with another field (as in `SubTask`) the `update` will raise `IntegrityError` when it increments `2` to `3`. The internal state of `to_shift` is now `3, 3, 4`, which is invalid.

If Django (and the db) supported `DEFERRED`, we could defer the unique check until the `update` had completed (`3, 4, 5`) and everything would work as in the case without `unique`.